botocore/utils.py

```diff
@@ -457,10 +459,12 @@ def _fetch_metadata_token(self):
                 if response.status_code == 200:
                     return response.text
                 elif response.status_code in (404, 403, 405):
+                    self._check_imdsv1_disabled()
```

Comment:
Instead of adding checks in all the failure cases, could we do this once to the beginning of `_get_request` similar to how we did `_assert_enabled`? That way we're sure we don't emit a request without a token and we avoid accidentally regressing this if a new failure path is added.

Reply:
This was the original design, but I misunderstood some feedback.  Just reverted it, thank you :)